### PR TITLE
shaderdb tests for GraphicsFuzz issues

### PR DESCRIPTION
**Pull request by afd**
Shaderdb tests for issues #922, #924, #926 and #927.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
Can one of the admins verify this patch?

**Comment by kuhar** [comment]
Thanks for the PR, @afd!
Seems like these three are passing (with and without assertions):
```
  shaderdb/fuzzer/GraphicsFuzz_TestOpCopyObjectFromAccessChain2.spvasm
  shaderdb/fuzzer/GraphicsFuzz_TestOpCopyObjectFromAccessChain3.spvasm
  shaderdb/fuzzer/GraphicsFuzz_TestOpPhisAtLoopHeader.spvasm
```

File: llpc/test/shaderdb/fuzzer/GraphicsFuzz_TestOpCopyObjectFromAccessChain2.spvasm [hunk 1]
```diff
@@ -0,0 +1,42 @@
+
```
Comment (amdrexu) [inline llpc/test/shaderdb/fuzzer/GraphicsFuzz_TestOpCopyObjectFromAccessChain2.spvasm:1]: Is it possible to rename the tests and remove the postfix "2". I don't want we have a bunch of tests that are XXX, XXX1, XXX2, XXX3. Meaningful names are more helpful.

File: llpc/test/shaderdb/fuzzer/GraphicsFuzz_TestOpCopyObjectFromAccessChain3.spvasm [hunk 2]
```diff
@@ -0,0 +1,36 @@
+
```
Comment (amdrexu) [inline llpc/test/shaderdb/fuzzer/GraphicsFuzz_TestOpCopyObjectFromAccessChain3.spvasm:1]: The same as above

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit d64d327edb9d11c0bdea8720310550381f908ed3
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 712f76fee75d69b23a1ea8f6465752c3ccaaf9a2</li>
<li>PAL: dev 9074bd0bdaf67fa84399dfabed3ec437951838f8</li>
<li>SPVGEN: dev c054813a9a894e32aaaa04c6717a667c15f60cfd</li>
<li>XGL: dev 598c6832a4983f5b75b38a589fca5be80a2f3bb0</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev ccbc834ac4f92e63a9a003bd5a18cb9fcf3c6121</li>
</ul></details>

<details><summary>CTS tests (Failed: 1/86494)</summary><ul><li>Built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix</li>
<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 22784/43245 (52.7%)</li>
<li>Failed: 0/43245 (0.0%)</li>
<li>Skipped: 20461/43245 (47.3%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 22782/43249 (52.7%)</li>
<li><details><summary>Failed: 1/43249 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.clipping.user_defined.clip_distance.vert.7
Stack trace: Flake
```
</p></details></li>

<li>Skipped: 20466/43249 (47.3%)</li>
</ul></details>

<details open><summary>Rhel 8.2, Gfx8</summary><ul>

<li>No Results</li>

</ul></details>


</ul></details>

File: llpc/test/shaderdb/fuzzer/GraphicsFuzz_TestOpCopyObjectFromAccessChain2.spvasm (hunk 1 above)
Comment (afd) [reply llpc/test/shaderdb/fuzzer/GraphicsFuzz_TestOpCopyObjectFromAccessChain2.spvasm:1]: I see your point.  However, what is meaningful about these tests is that they exposed crashes in LLPC - the represent the minimized SPIR-V that could reproduce the associated crash.  Beyond that, the specific SPIR-V that they contain isn't particularly meaningful.

I can see two options:

(1) Give them a name inspired by the instructions that remain after minimisation, e.g. "ExtractAndCopyObject" if the test involves a combination of OpCompositeExtract and OpCopyObject.  

(2) Give them a name containing details of the problem that they triggered, e.g. "InlineAttributeCompatibilityProblem" for #923.

What do you think would be best, @amdrexu ?

**Comment by afd** [comment]
@kuhar Regarding the tests that were passing: it turns out I had not been using the latest dev commit, so I found a few issues that must have been fixed by recent patches.

The tests that pass could thus be kept for the sake of thoroughness (i.e., they did trigger something interesting several commits back), or discarded; I don't mind either way.

File: llpc/test/shaderdb/fuzzer/GraphicsFuzz_TestOpCopyObjectFromAccessChain2.spvasm (hunk 1 above)
Comment (amdrexu) [reply llpc/test/shaderdb/fuzzer/GraphicsFuzz_TestOpCopyObjectFromAccessChain2.spvasm:1]: Thank you for explanations. I think option 2 is more reasonable, such as XXX_TestInlineAttributeCompatibility. I think the postfix "problem" is unnecessary. All fuzz tests are to catch problems :)

**Comment by kuhar** [comment]
@afd Since these were legitimate failure at some point, I'd vote for keeping un-XFAILing these and keeping them as regression tests. The maintenance cost seems minimal to me.

**Comment by JaxLinAMD** [comment]
@afd  please rebase to lastest dev to make jenkins build pass

File: llpc/test/shaderdb/fuzzer/GraphicsFuzz_TestOpCopyObjectFromAccessChain2.spvasm (hunk 1 above)
Comment (afd) [reply llpc/test/shaderdb/fuzzer/GraphicsFuzz_TestOpCopyObjectFromAccessChain2.spvasm]: I have renamed the tests.

File: llpc/test/shaderdb/fuzzer/GraphicsFuzz_TestOpCopyObjectFromAccessChain3.spvasm (hunk 2 above)
Comment (afd) [reply llpc/test/shaderdb/fuzzer/GraphicsFuzz_TestOpCopyObjectFromAccessChain3.spvasm]: Done.

**Comment by afd** [comment]
@JaxLinAMD done.

**Review by amdrexu — APPROVED** [review]

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by JaxLinAMD** [comment]
retest this please

**Pull request merged**